src/jit/flowgraph.cpp

```diff
@@ -11002,12 +11002,12 @@ void Compiler::fgUnlinkRange(BasicBlock* bBeg, BasicBlock* bEnd)
 
 void Compiler::fgRemoveBlock(BasicBlock* block, bool unreachable)
 {
-    BasicBlock* bPrev = block->bbPrev;
-
     /* The block has to be either unreachable or empty */
 
     PREFIX_ASSUME(block != nullptr);
```

Comment:
@janvorli Does it make sense to use `PREFIX_ASSUME` for null checks? I think in debug it is the same behavior as a regular assert and in release it can't give any valuable hint for the compiler, can it?

Reply:
In release, it is expanded to `__assume`. In static analyzer builds, it expands to `if (!(_condition)) __UNREACHABLE();` in all build configurations. So it seems it makes sense.

Reply:
I mean the release code for:
```
__assume(p != nullptr);
p->field;
```

and for
```
p->field;
```
would be the same.

Reply:
I am not sure if I understand - the __assume is a hint to the compiler. Of course the generated code would be the same. So the PREFIX_ASSUME is basically an extended kind of assert that static analysis tools can understand even in release build.

Reply:
Thanks Jan.